Alg.java

```diff
@@ -0,0 +1,605 @@
+      else if(g1.equals("Jazz")) {
+         if(g2.equals("Blues"))
+            return 17;
+         else if(g2.equals("Rock"))
+            return 12;
+         else if(g2.equals("Rap"))
+            return 15;
+      }
+      else if(g1.equals("Latin")) {
+         if(g2.equals("World"))
+            return 12;
+         else if(g2.equals("Jazz"))
+            return 10;
+         else if(g2.equals("Pop"))
+            return 10;
+         else if(g2.equals("Rap"))
+            return 13;
+         else if(g2.equals("RnB"))
+            return 10;
+      }
+      else if(g1.equals("Metal")) {
+         if(g2.equals("Electronic"))
+            return 10;
+         else if(g2.equals("Punk"))
+            return 15;
+         else if(g2.equals("Rock"))
+            return 13;
+      }
+      else if(g1.equals("Pop")) {
+         if(g2.equals("Country"))
+            return 8;
+         else if(g2.equals("Latin"))
+            return 10;
+         else if(g2.equals("Rap"))
+            return 12;
+         else if(g2.equals("RnB"))
+            return 15;
+         else if(g2.equals("World"))
+            return 10;
+      }
+      else if(g1.equals("Punk")) {
+         if(g2.equals("Metal"))
+            return 15;
+         else if(g2.equals("Rock"))
+            return 15;
+      }
+      else if(g1.equals("Rap")) {
+         if(g2.equals("Electronic"))
+            return 10;
+         else if(g2.equals("Jazz"))
+            return 15;
+         else if(g2.equals("Latin"))
+            return 13;
+         else if(g2.equals("RnB"))
+            return 15;
+      }
+      else if(g1.equals("Reggae")) {
+         if(g2.equals("Blues"))
+            return 10;
+         else if(g2.equals("Electronic"))
+            return 8;
+         else if(g2.equals("Rock"))
+            return 12;
+      }
+      else if(g1.equals("RnB")) {
+         if(g2.equals("Pop"))
+            return 15;
+         else if(g2.equals("Rap"))
+            return 15;
+         else if(g2.equals("Latin"))
+            return 10;
+         else if(g2.equals("World"))
+            return 12;
+     }
+     else if(g1.equals("Rock")) {
+        if(g2.equals("Country"))
+           return 10;
+        else if(g2.equals("Folk"))
+           return 10;
+        else if(g2.equals("Jazz"))
+           return 12;
+        else if(g2.equals("Metal"))
+           return 13;
+        else if(g2.equals("Punk"))
+           return 15;
+        else if(g2.equals("Blues"))
+           return 13;
+     }
+     else if(g1.equals("World")) {
+        if(g2.equals("Latin"))
+           return 13;
+        else if(g2.equals("Pop"))
+           return 10;
+        else if(g2.equals("Country"))
+           return 10;
+        else if(g2.equals("RnB"))
+           return 12;
+     }
+     return 0;
+   }
+
+   public PriorityQueue<SongWrapper> matchingAlg(Song test, int[] tags) {
+      PriorityQueue<SongWrapper> queue = new PriorityQueue<SongWrapper>(25);
+      for(int i = 0; i < 1268; i++) {
```

Comment:
magic numbers!!!

Look at the SongDatabaseDriver class: there's a getAllSongs() method that returns the primary key of every song in the database.